devtools/check_unmarked_tests.sh

```diff
@@ -0,0 +1,11 @@
+#!/bin/sh
+
+unmarked=$(pytest tests/ --collect-only -m "not unit and not regression" -q | head -n -2) 2> /dev/null
```

Comment:
This will check every file in tests folder every time we add a commit, right? I guess there is a way to just check the changed files. 

```bash
#!/bin/bash

# Start the timer
start_time=$SECONDS

# Initialize an empty variable to accumulate unmarked tests
unmarked=""

# Loop through each provided file
for file in "$@"; do
  # Check if the file is a Python file in the tests directory
  if [[ "$file" == tests/*.py ]]; then
    echo "checking tests in $file"
    # Collect unmarked tests for the specific file
    file_unmarked=$(pytest "$file" --collect-only -m "not unit and not regression" -q | head -n -2) 2> /dev/null
    if [[ -n "$file_unmarked" ]]; then
      unmarked+="$file_unmarked"$'\n'
    fi
  fi
done

# Count the number of unmarked tests found
num_unmarked=$(echo "$unmarked" | sed '/^\s*$/d' | wc -l)

# Calculate the elapsed time
elapsed_time=$(($SECONDS - $start_time))
# If there are any unmarked tests, print them and exit with status 1
if [ "$num_unmarked" -gt 0 ]; then
    echo "Time taken: ${elapsed_time} seconds"
    echo "----found unmarked tests----"
    echo "$unmarked"
    exit 1
fi
```

Also, need to set `pass_filenames` to `true` in config yaml

Reply:
yeah but most of the time is spent on pytest init, regardless of how many files are changed, so checking everything isn't any more expensive

Reply:
Yes, but this change reduced it from 8 seconds to 4 seconds on my laptop.

Reply:
feel free to change it if you want, i don't have any strong preference.